libkbfs/block_types.go

```diff
@@ -188,10 +188,30 @@ func (fb *FileBlock) NewEmpty() Block {
 
 // DataVersion returns data version for this block.
 func (fb *FileBlock) DataVersion() DataVer {
+	// If this is an indirect block, and none of its children are
+	// marked as direct blocks, then this must be a big file.  Note
+	// that we do it this way, rather than returning on the first
+	// non-direct block, to support appending to existing files and
```

Comment:
Ah, right, I see. That raises another question, though: should `DirectType` be part of `BlockInfo` instead of `BlockPointer`? I think I previously convinced myself that it had to be in `BlockPointer`, but now I can't remember why. Perhaps that deserves a comment above the `DirectType` field in `BlockPointer`? And a comment/TODO re. the problem you mentioned.

I guess if we only set `DirectType` for new `BlockPointer`s, the panics in `DataVersion` can look like:

```
if all IPtrs in fb.IPtrs has unknown direct type, panic
if there are multiple known values of IPtr.DirectType, panic
```